lib/mixins/form.js

```diff
@@ -1,17 +1,34 @@
 export default {
+    computed: {
+        inputClass() {
+            return [
+                this.size ? `form-control-${this.size}` : null,
+                this.state ? `form-control-${this.state}` : null
+            ];
+        },
+        custom() {
+            return !this.plain;
+        }
+    },
     props: {
         name: {
             type: String
         },
-        id: {
-            type: String
-        },
         disabled: {
             type: Boolean
         },
-        required: {
+        plain: {
             type: Boolean,
             default: false
+        },
+        state: {
+            type: String
+        },
+        size: {
+            type: String
```

Comment:
And state and size should not be in form.js any more either, as they have been moved over to the components that only need it.